packages/graphql/src/schema/augment/fulltext.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { SchemaComposer } from "graphql-compose";
+import type { Node } from "../../classes";
+import { fulltextResolver } from "../resolvers/query/fulltext";
+import { upperFirst } from "../../utils/upper-first";
+import { FloatWhere } from "../../graphql/input-objects/FloatWhere";
+import { SCORE_FIELD } from "../../graphql/directives/fulltext";
+
+export function augmentFulltextSchema(
+    node: Node,
+    composer: SchemaComposer,
+    nodeWhereTypeName: string,
+    nodeSortTypeName: string
+) {
+    if (node.fulltextDirective) {
+        const fields = node.fulltextDirective.indexes.reduce(
+            (res, index) => ({
+                ...res,
+                [index.name]: composer.createInputTC({
+                    name: `${node.name}${upperFirst(index.name)}Fulltext`,
+                    fields: {
+                        phrase: "String!",
+                    },
+                }),
+            }),
+            {}
+        );
+
+        const fulltextResultDescription = `The result of a fulltext search on an index of ${node.name}`;
+        const fulltextWhereDescription = `The input for filtering a fulltext query on an index of ${node.name}`;
+        const fulltextSortDescription = `The input for sorting a fulltext query on an index of ${node.name}`;
+
+        composer.createInputTC({
+            name: `${node.name}Fulltext`,
+            fields,
+        });
+
+        composer.createInputTC({
+            name: node.fulltextTypeNames.sort,
+            description: fulltextSortDescription,
+            fields: {
+                [SCORE_FIELD]: "SortDirection",
+                [node.singular]: nodeSortTypeName,
+            },
+        });
+
+        composer.createInputTC({
+            name: node.fulltextTypeNames.where,
+            description: fulltextWhereDescription,
+            fields: {
+                [SCORE_FIELD]: FloatWhere.name,
+                [node.singular]: nodeWhereTypeName,
+            },
+        });
+
+        composer.createObjectTC({
+            name: node.fulltextTypeNames.result,
+            description: fulltextResultDescription,
+            fields: {
+                [SCORE_FIELD]: "Float!",
```

Comment:
Just a personal preference when using `graphql-compose` - I prefer to use type objects here, for example, this would be `new GraphQLNonNull(GraphQLFloat)`. I think it just makes it easier to find all of the instances of a particular type in our schema augmentation, but it all works the same!

Same for our own generated types when we use them elsewhere, best to reference the objects themselves because then if we change the name or something, it doesn't all break.